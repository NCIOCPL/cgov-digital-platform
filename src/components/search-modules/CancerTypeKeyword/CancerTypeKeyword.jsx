import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Fieldset, Autocomplete } from '../../atomic';
import { getDiseasesForSimpleTypeAhead } from '../../../store/actions';

const CancerTypeKeyword = ({ handleUpdate }) => {
  const dispatch = useDispatch();
  const { keywordPhrases, cancerType } = useSelector(store => store.form);
  const { diseases = [] } = useSelector(store => store.cache);

  const [CTK, setCTK] = useState({ value: (cancerType.name)? cancerType.name : keywordPhrases });

  useEffect(() => {
    dispatch(getDiseasesForSimpleTypeAhead({name: CTK.value}));
  }, [CTK, dispatch]);

  const matchItemToTerm = (item, value) => {
    return item.name.toLowerCase().indexOf(value.toLowerCase()) !== -1;
  };

  // when selecting from dropdown, make it the cancer type and clear keywords/phrases
  const handleSelection = (diseaseResult) => {
    handleUpdate('cancerType', diseaseResult);
    handleUpdate('keywordPhrases', '');
  }

  return (
    <Fieldset
      id="type"
      legend="Cancer Type/Keyword"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
    >
      <Autocomplete
        id="ctk"
        label="Cancer Type/Keyword"
        value={CTK.value}
        inputProps={{ placeholder: 'Start typing to select a cancer type or keyword' }}
        wrapperStyle={{ position: 'relative', display: 'inline-block' }}
        items={diseases}
        inputHelpText="Leave blank to search all cancer types or keywords."
        getItemValue={item => item.name}
        shouldItemRender={matchItemToTerm}
        onChange={(event, value) => {
          setCTK({ value });
          handleUpdate('cancerType', {name: '', codes: []})
          handleUpdate('keywordPhrases', value);
          handleUpdate('typeCode', {});
        }}
        onSelect={(value, item) => {
          setCTK({ value });
          handleUpdate('typeCode', item);
          handleUpdate('keywordPhrases', value);
        }}
        renderMenu={children => (
          <div className="cts-autocomplete__menu --q">
            {children.length ? (
                  children
                ) : (
                  <div className="cts-autocomplete__menu-item">
                    No available options found.  Your search will be based on the text above.
                  </div>
                )}
          </div>
        )}
        renderItem={(item, isHighlighted) => (
          <div
            className={`cts-autocomplete__menu-item ${
              isHighlighted ? 'highlighted' : ''
            }`}
            key={item.codes[0]}
          >
            {item.name} {item.ancestor_ids} {item.parent_ids} {item.type}
          </div>
        )}
      />
    </Fieldset>
  );
};

export default CancerTypeKeyword;
