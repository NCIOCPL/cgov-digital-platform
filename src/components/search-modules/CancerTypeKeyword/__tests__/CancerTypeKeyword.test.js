import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import CancerTypeKeyword from '../CancerTypeKeyword';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <CancerTypeKeyword {...props} />
  );
  return component;
};

describe('CancerTypeKeyword', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
