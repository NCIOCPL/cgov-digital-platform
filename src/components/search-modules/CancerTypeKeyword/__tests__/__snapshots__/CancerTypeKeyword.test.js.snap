// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancerTypeKeyword Render renders without error 1`] = `
<fieldset
  class="cts-fieldset "
  id="fieldset--type"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Cancer Type/Keyword
    </span>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
  </legend>
  <div
    class="cts-fieldset__body"
  >
    <div
      class="cts-input-group "
    >
      <input
        aria-label="Cancer Type/Keyword"
        aria-required="false"
        class="cts-input  "
        id="q"
        name="CancerTypeKeyword"
        placeholder="Start typing to select a cancer type"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</fieldset>
`;
